### ISSUE TITLE
Reassess the use of the "surveillance" personality on fleets

### ISSUE THREAD

**Issue by Amazinite** [endless-sky/endless-sky#9894]
### Problem Description

Quoting from #9714,
> I'm pretty sure we're overusing the surveillance personality. A ship does not need the surveillance personality in order to scan other ships. All it needs is an outfit or cargo scanner and no hostiles in the system to distract it. The surveillance personality was always a way to make such ships more aggressive in their scanning, hence them scanning every ship personally instead of not scanning ships their government has already scanned. I've maintained the more aggressive scanning behavior of the surveillance personality by giving them a higher threshold for how many ships they can scan, for how long, and how they choose their scan targets, but we may still want to reevaluate the fleets we have given the surveillance personality.

The surveillance personality seems extremely tailored to the story of the main human campaign. The surveillance personality, prior to #9714, can be summarized as doing three things:
1. Surveillance ships will personally scan every ship in the system if possible, instead of not scanning ships that have already been scanned by the surveillance ship's government.
2. Surveillance ships will ditch their fighters in the system before leaving.
3. Surveillance ships will "scan" stellar objects in the system if they have the "atmosphere scan" attribute.

The latter two behaviors are entirely designed around having Navy surveillance fleets ditch surveillance drones for the player to capture in that one Free Worlds intro mission, and also tie into the story of the Navy searching planets for signs of nuclear testing.

Among randomly spawned fleets, the surveillance personality is used by the following:
* Navy Surveillance
  * Uses all aspects of the surveillance personality, as it was literally made for this fleet.
* Hai Surveillance
  * These fleets have no fighters or atmosphere scan, and so don't use most of the surveillance personality.
* Remnant Home Guard
  * This fleet is only fighters; Terns, Petrels, and Smews. Only Terns have atmosphere scan.
  * This fleet has "surveillance" and "mining", but the mining behavior will never run because it's overtaken by the surveillance personality.
* Korath Surveillance
  * This fleet has fighters, and those fighters have atmosphere scan.
  * The fighters also have cargo scanning, which means they could conceivably fine you if ever you're friendly with them. This probably shouldn't be the case; we should set the government's fine modifier to 0.
* Vyrmeid (Plankton)
  * Only used to make atmosphere scan change the ship behavior.
* Wanderer Drones
  * Only used to make atmosphere scan change the ship behavior.
* Embershades
  * Only used to make atmosphere scan change the ship behavior.
* Iije (Surveillance)
  * Only used to make atmosphere scan change the ship behavior.

### Related Issue Links

#9714
#9706

### Desired Solution

It seems like there are two reasons that the surveillance personality is being used:
* The fleet is intended to scan the player, so the fleet creator just threw the surveillance personality onto the fleet, even if it wasn't actually necessary to do so.
* The fleet needs some sort of lingering behavior, so the ships were given atmosphere scan and surveillance so that they actually do something instead of just jumping around.

As such, I believe the following changes should be made:
* For the cases where surveillance is used to make ships linger in the system, just give them the lingering personality.
  * Lingering ships will simply try to stay near the system center, so perhaps we should give lingering/staying ships more behaviors, potentially personality dependent.
* Potentially split the surveillance personality up, or create new personalities that do only part of what surveillance does. For example, the atmosphere scan lingering behavior could be its own personality, as well as the fighter dumping behavior.

### Alternative Approaches

N/A

### Additional Context

Why is Ship::Linger a thing? Seems like there should just be a map<const Ship *, int> in AI to count this.

**Comment by Amazinite** [comment]
That's simply how it's coded. The atmosphere scan attribute isn't used anywhere else aside from in the surveillance personality logic. There's no particular reason it needs to remain that way, though.

**Comment by ziproot** [comment]
tl;dr: Every fleet that has the surveillance personality also has a ship with atmosphere scan, with one exception, the Vyrmeid, though Hai Surveillance only gets one during Hai Reveal. There are many fleets that have ships with atmosphere scan but do not have the surveillance personality.

NOTE: Vyrmeid has "atmospheric scan" which does nothing. I think this is a bug.

**Fleets with atmosphere scan and no surveillance**
Large Kimek
Large Saryd
Saryd Miners
Anti-Pirate Hai (Wormhole Travel)
Anti-Pirate Hai (No Wormhole Travel)
Small Hai Merchant (Human)*
Large Hai Merchant (Human)*
Iije (Uninterested)
Iije (Swarming)
Korsmanath A'awojs
Korath Civilian Miners
Korath Civilian Large
Small Remnant
Large Remnant
Remnant Transport
Light Remnant Defense
Heavy Remnant Defense
Wanderer Defense
Wanderer Kor Efret Traders

**Persons with atmosphere scan and no surveillance**
Prototype B3-CC4
MasterOfGrey
Zitchas

**Fleets with surveillance and no atmosphere scan**
Vyrmeid (Plankton)

**Persons with surveillance and no atmosphere scan**
Cap'n Pester
Marauding Max
Local God

**Fleets with atmosphere scan and surveillance**
Embershades
Hai Surveillance†
Iije (Surveillance)
Korath Surveillance
Korath Rescue
Remnant Home Guard
Wanderer Drones
Navy Surveillance that won't enter the wormhole
Navy Surveillance

* 20-30 days after completing Hai Leaks Respone 4
† 32-42 days after completing Hai Building Mountaintop [3] Furnishings
